test/org/sagebionetworks/bridge/play/controllers/BaseControllerIntegTest.java

```diff
@@ -0,0 +1,88 @@
+package org.sagebionetworks.bridge.play.controllers;
+
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.sagebionetworks.bridge.TestUserAdminHelper;
+import org.sagebionetworks.bridge.TestUtils;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import play.libs.ws.WS;
+import play.libs.ws.WSRequest;
+import play.libs.ws.WSResponse;
+
+import javax.annotation.Resource;
+
+import static org.apache.http.HttpHeaders.ACCEPT_LANGUAGE;
+import static org.apache.http.HttpHeaders.USER_AGENT;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import static org.sagebionetworks.bridge.BridgeConstants.BRIDGE_API_STATUS_HEADER;
+import static org.sagebionetworks.bridge.BridgeConstants.BRIDGE_WARNING_STATUS;
+import static org.sagebionetworks.bridge.TestConstants.TEST_BASE_URL;
+import static org.sagebionetworks.bridge.TestConstants.TIMEOUT;
+
+@ContextConfiguration("classpath:test-context.xml")
+@RunWith(SpringJUnit4ClassRunner.class)
+public class BaseControllerIntegTest {
+
+    @Resource
+    private TestUserAdminHelper helper;
+
+    private TestUserAdminHelper.TestUser testUser;
+
+    @Before
+    public void before() {
+        testUser = helper.getBuilder(BaseControllerIntegTest.class).build();
+    }
+
+    @After
+    public void after() {
+        helper.deleteUser(testUser);
+    }
+
+    @Test
+    public void setWarningHeaderWhenNoUserAgentOrNoAcceptLanguage() {
+        // This is expensive so test them all after starting the server, in one test
+        TestUtils.runningTestServerWithSpring(() -> {
```

Comment:
We want to avoid writing any more integration tests in BridgePF. Please write a test that verifies the controller sets the headers, that does not start a Play server. If we write a true integration test, we can do it in the integration tests project. This kind of test is extremely slow to run.